spec/integration/support/serve_examples.rb

```diff
@@ -74,6 +74,54 @@ def test_basic_html(port)
       expect(curl_command).to succeed.with_stderr.and have_stdout(expected_content)
     end
 
+    it "prints a warning when --public is used" do
+      cmd = "#{machinery_command} serve opensuse131 --port 5000 --public"
+      res = nil
+      Thread.new do
+        res = @machinery.run_command(cmd, stderr: :capture)
+      end
+
+      test_basic_html(5000)
+
+      @machinery.run_command("pkill -f 'machinery serve' --signal 15")
+
+      # wait until the process quits and writes the response of the program
+      # into 'res'
+      times = 0
+      while res.nil? && times < 10
+        times += 1
+        sleep(1)
+      end
+
+      expect(res.stderr).to include("The --public option makes the HTTP server listen on all " \
```

Comment:
This is again too specific, I'd only check for `--public option` or so, so that the test doesn't fail if small details of the error message are changed.